lxd/container_lxc.go

```diff
@@ -3561,16 +3561,15 @@ func (c *containerLXC) Delete() error {
 				// Remove snapshot volume and database record.
 				err = pool.DeleteInstanceSnapshot(c, nil)
 				if err != nil {
```

Comment:
So this may leave stuff laying around which isn't so good.

Any chance we can return a standard error instead and only skip if it's that error?
Similar to how your can use ErrNotExist or similar for files.

Reply:
@stgraber yep makes sense, will add an ErrNotExist error as suggested, and if that is detected then skip otherwise fail. Will port this to the VM instance type's Delete() function too.

Reply:
@stgraber ive removed this commit from this PR and will do the larger change in a separate PR.